pkg/beholder/client_test.go

```diff
@@ -0,0 +1,353 @@
+							assert.True(t, equal, fmt.Sprintf("Record attributes mismatch for key %v", key))
+							return true
+						})
+						for key := range customAttributes {
+							if _, ok := actualAttributeKeys[key]; !ok {
+								t.Fatalf("Record attribute key not found: %s", key)
+							}
+						}
+						done <- struct{}{}
+					})
+			}
+			for i := 0; i < tc.messageCount; i++ {
+				go tc.messageGenerator(client, tc.messageBody, customAttributes)
+			}
+
+			select {
+			case <-done:
+			case <-time.After(10 * time.Second):
+				t.Fatalf("Timed out waiting for messages to be emitted")
+			}
+		})
+	}
+}
+
+func TestEmitterMessageValidation(t *testing.T) {
+	getEmitter := func(exporterMock *mocks.OTLPExporter) Emitter {
+		client, err := newOtelClient(
+			DefaultConfig(),
+			func(err error) { t.Fatalf("otel error: %v", err) },
+			// Override exporter factory which is used by Client
+			func(context.Context, ...otlploggrpc.Option) (sdklog.Exporter, error) {
+				return exporterMock, nil
+			},
+		)
+		assert.NoError(t, err)
+		return client.Emitter
+	}
+
+	for _, tc := range []struct {
+		name                string
+		attrs               Attributes
+		exporterCalledTimes int
+		expectedError       string
+	}{
+		{
+			name: "Missing required attribute",
+			attrs: Attributes{
+				"key": "value",
+			},
+			exporterCalledTimes: 0,
+			expectedError:       "'Metadata.BeholderDataSchema' Error:Field validation for 'BeholderDataSchema' failed on the 'required' tag",
+		},
+		{
+			name: "Invalid URI",
+			attrs: Attributes{
+				"beholder_data_schema": "example-schema",
+			},
+			exporterCalledTimes: 0,
+			expectedError:       "'Metadata.BeholderDataSchema' Error:Field validation for 'BeholderDataSchema' failed on the 'uri' tag",
+		},
+		{
+			name:                "Valid URI",
+			exporterCalledTimes: 1,
+			attrs: Attributes{
+				"beholder_data_schema": "/example-schema/versions/1",
+			},
+			expectedError: "",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			setupMock := func(exporterMock *mocks.OTLPExporter) (*mocks.OTLPExporter, <-chan struct{}) {
+				done := make(chan struct{}, tc.exporterCalledTimes)
+				if tc.exporterCalledTimes > 0 {
+					exporterMock.On("Export", mock.Anything, mock.Anything).Return(nil).Times(tc.exporterCalledTimes).
+						Run(func(args mock.Arguments) {
+							done <- struct{}{}
+						})
+				}
+				return exporterMock, done
+			}
+
+			assertError := func(err error, expected string) {
+				if tc.expectedError != "" {
+					assert.ErrorContains(t, err, expected)
+				} else {
+					assert.NoError(t, err)
+				}
+			}
+
+			assertMock := func(exporterMock *mocks.OTLPExporter) {
+				if tc.exporterCalledTimes > 0 {
+					exporterMock.AssertExpectations(t)
+				} else {
+					exporterMock.AssertNotCalled(t, "Export")
+				}
+			}
+
+			waitUntilSent := func(done <-chan struct{}) {
+				for range tc.exporterCalledTimes {
+					select {
+					case <-done:
+					case <-time.After(10 * time.Second):
```

Comment:
is there a way to instantly flush messages?

Reply:
Addressed https://github.com/smartcontractkit/chainlink-common/pull/694/commits/c8e8291c0792cd0d2f6f4cca0478361477481dd8